QD-664 view generated results (#17)
JetBrains/Qodana: General/output.md

- **log/**  - contains  `idea.log`, `gradle.log`, and so on.
- **projectStructure/** - Metainformation about your project: modules, frameworks/libraries, roots, and so on.
- **report/** - HTML report (generated via `--save-report`/`--show-report`)
- **profile.xml** - effective profile in the IDEA format: all inspections and their configurations. Depends on the configured profile, `qodana.yaml`, and the plugin list.
- **.descriptions.json** - descriptions for all inspections.
- **%InspectionId%.json** - contains all problems generated by the inspection with this ID: `%InspectionId%` message, place, severity, and so on.

## UI-compatible output

> In addition to programmatic output you can also generate human readable output in HTML format (`--save-report`). Unfortunately due to js security restrictions this cannot be currently viewed via file:// protocol (double-click on index.html). For convenience, we provide `--show-report` argument to serve HTML report via http:// protol locally.  

Reviewer: ```suggestion
In addition to programmatic output, you can generate human readable output in the HTML format by using the `--save-report` argument. 

### Local run

Due to JavaScript security restrictions, the generated report cannot be viewed via the `file://` protocol (that is, by double-clicking the `index.html` file). Instead, you can use the `--show-report` argument to serve the HTML report locally via the `http://` protocol.
```

> In case you want to view UI in already generated `report/` folder:

Reviewer: ```suggestion
To view the report in the already generated `report/` folder:
```

 - Docker
    ```
    docker run -it --rm -p 8000:80 -v $(pwd)/report:/usr/share/nginx/html nginx
    ```  
 - Python  
    If you have python installed, you can serve current folder content via:
    ```
    cd report/
    python2 -m SimpleHTTPServer
    # or
    python3 -m http.server

>     ```

Reviewer: Let's add PHP here, too?

```suggestion
    ```

- PHP
    If you have PHP installed, you can serve current folder content via:
    ```
    php -S localhost:8000
    ```
```

Report would be available at [http://localhost:8000](http://localhost:8000), you can stop the web server by pressing `Ctrl-C`.

> In case of [GitLab CI](https://github.com/JetBrains/Qodana/tree/main/Docker#quick-start-with-recommended-profile) you can directly view the report by opening `report/index.html` from artifacts. For [Github Action](https://github.com/JetBrains/Qodana/tree/main/Docker#quick-start-with-recommended-profile) consider uploading artifacts to some [s3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/static-website-hosting.html) or using [Qodana Github App](https://github.com/JetBrains/Qodana/blob/main/GitHub/README.md#qodana-github-app) instead.

Reviewer: ```suggestion
### GitLab CI

In case of [GitLab CI](https://github.com/JetBrains/Qodana/tree/main/Docker#quick-start-with-recommended-profile), you can directly view the report by opening `report/index.html` from artifacts. 

### Github action

For [Github Action](https://github.com/JetBrains/Qodana/tree/main/Docker#quick-start-with-recommended-profile), consider uploading artifacts to some [s3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/static-website-hosting.html) or using [Qodana Github App](https://github.com/JetBrains/Qodana/blob/main/GitHub/README.md#qodana-github-app) instead.
```
